src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/organisations/OrganisationsService.kt

```diff
@@ -62,29 +66,77 @@ class OrganisationsService(
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  suspend fun organisationCreated(event: OrganisationEvent) = log.debug("Received organisation created event for organisation {}", event.organisationId)
-  suspend fun organisationUpdated(event: OrganisationEvent) = log.debug("Received organisation updated event for organisation {}", event.organisationId)
+  suspend fun organisationCreated(event: OrganisationEvent) {
+    val entityName = ORGANISATION.entityName
+    val dpsOrganisationId = event.organisationId
+    val telemetryMap = mutableMapOf(
+      "organisationId" to dpsOrganisationId.toString(),
+    )
 
-  suspend fun organisationDeleted(event: OrganisationDeletedEvent) {
-    log.debug("Received organisation deleted event for organisation {}", event.organisationId)
-    val dpsOrganisationId = event.organisationId.toString()
-    val telemetryMap = mutableMapOf("dpsOrganisationId" to dpsOrganisationId)
     if (event.didOriginateInDPS()) {
-      runCatching {
-        mappingApiService.getByDpsOrganisationIdOrNull(dpsOrganisationId)?.also { mapping ->
-          telemetryMap["nomisOrganisationId"] = mapping.nomisId.toString()
-          nomisApiService.deleteCorporateOrganisation(corporateId = mapping.nomisId)
-          tryToDeleteMapping(dpsOrganisationId)
-          telemetryClient.trackEvent("organisation-deleted-success", telemetryMap)
-        } ?: also {
-          telemetryClient.trackEvent("organisation-deleted-skipped", telemetryMap)
+      synchronise {
+        name = entityName
+        telemetryClient = this@OrganisationsService.telemetryClient
+        retryQueueService = organisationsRetryQueueService
+        eventTelemetry = telemetryMap
+
+        checkMappingDoesNotExist {
+          mappingApiService.getByDpsOrganisationIdOrNull(dpsOrganisationId)
```

Comment:
technically I don't need to go to the mapping service for organisations, but seemed easier to do so to ensure that we don't call nomis prisoner api to do a create when it already exists etc.